staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -96,6 +96,13 @@ type ExtraConfig struct {
 	ServiceResolver ServiceResolver
 
 	RejectForwardingRedirects bool
+
+	// DisableAvailableConditionController disables the controller that updates the Available conditions for
+	// APIServices, Endpoints and Services. This controller runs in kube-aggregator and can interfere with
+	// Generic Control Plane components when certain apis are not available.
+	// TODO: We should find a better way to handle this. For now it will be for Generic Control Plane authors to
+	// disable this controller if they see issues.
+	DisableAvailableConditionController bool
```

Comment:
Doesn't this mean that all APIService are unavailable then?

Reply:
If I read code right it will impact only apiservices which uses services (not Local ones) as they need service and endpoint objects:
```
if apiService.Spec.Service == nil {
			continue
		}
```
This is one of the reasons I was on the line how to fix this issue in generic-controlplane. Its very edge condition when you need this controller: If want to use Services and Endpoints only. And I'm not quite fully sure if configuration where you have those but not APIServices is feasible too. I suspect no.